embed/etcd.go

```diff
@@ -424,18 +424,20 @@ func (e *Etcd) Close() {
 }
 
 func stopServers(ctx context.Context, ss *servers) {
-	shutdownNow := func() {
-		// first, close the http.Server
+	// first, close the http.Server
+	if ss.http != nil {
 		ss.http.Shutdown(ctx)
-		// then close grpc.Server; cancels all active RPCs
-		ss.grpc.Stop()
+	}
+
+	if ss.grpc == nil {
+		return
 	}
```

Comment:
It seems that the change wasn't backported to 3.4 previously?

Reply:
It seems we lost this patch. I think it's caused by different code base.

Reply:
could you link this PR to the original patch (PR)?

Reply:
https://github.com/etcd-io/etcd/commit/bf12179a5ad97710e40c056fe7e9b600f7f981e2#diff-19e9a359c00adb10524edbf94654438284606cb1e518e0f550ceee8376f7277f

[#15446](https://github.com/etcd-io/etcd/pull/15446)

Reply:
We have to stop http server as first, which also applies to http-only server. After we stop the http server, the [serversC channel](https://github.com/etcd-io/etcd/blob/4e800bdc6613cce4c67203b344a8959950ea1e3b/embed/serve.go#L115) can be stopped. And the channel range can be stopped.

https://github.com/etcd-io/etcd/blob/4e800bdc6613cce4c67203b344a8959950ea1e3b/embed/etcd.go#L389-L394

Otherwise, the server will be stuck at terminating status forever. This bug only happens in http-only insecure mode.
So I follow the main branch's pattern and enable graceful-stop to test it.